# Machine Learning Lab-2 (100 points)

## 1) Load the Diabetes dataset into a dataframe (50 points)

You can find the diabetes dataset at https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_diabetes.html#sklearn.datasets.load_diabetes.
You can use the Boston dataset file as a reference

### 1.1) Load the dataset (10 points)

In [2]:
from sklearn.datasets import load_diabetes
diabetes_data=load_diabetes()

### 1.2) Print a description of the data (10 points)

In [3]:
print(diabetes_data.DESCR)


.. _diabetes_dataset:

Diabetes dataset
----------------

Ten baseline variables, age, sex, body mass index, average blood
pressure, and six blood serum measurements were obtained for each of n =
442 diabetes patients, as well as the response of interest, a
quantitative measure of disease progression one year after baseline.

**Data Set Characteristics:**

  :Number of Instances: 442

  :Number of Attributes: First 10 columns are numeric predictive values

  :Target: Column 11 is a quantitative measure of disease progression one year after baseline

  :Attribute Information:
      - age     age in years
      - sex
      - bmi     body mass index
      - bp      average blood pressure
      - s1      tc, T-Cells (a type of white blood cells)
      - s2      ldl, low-density lipoproteins
      - s3      hdl, high-density lipoproteins
      - s4      tch, thyroid stimulating hormone
      - s5      ltg, lamotrigine
      - s6      glu, blood sugar level

Note: Each of these 10 feature va

### 1.3) Write a small description of the Diabetes dataset. You can use the output of the previous cell for your reference (10 points)

#### *The Diabetes DataSet consists of  442 rows and  10 columns.* 
#### *The target column is the 11-th column which is quantitative and measures disease progression.*
#### *All attributes are numeric ("sex" attribute is either 1 or 2).*  

### 1.4) Create a dataframe and import the diabetes data (input) to the dataframe. Print the dataframe that you have created (10 points)

In [6]:
import pandas as pd
df=pd.DataFrame(data=diabetes_data.data, columns=diabetes_data.feature_names)
print(df)

          age       sex       bmi        bp        s1        s2        s3  \
0    0.038076  0.050680  0.061696  0.021872 -0.044223 -0.034821 -0.043401   
1   -0.001882 -0.044642 -0.051474 -0.026328 -0.008449 -0.019163  0.074412   
2    0.085299  0.050680  0.044451 -0.005671 -0.045599 -0.034194 -0.032356   
3   -0.089063 -0.044642 -0.011595 -0.036656  0.012191  0.024991 -0.036038   
4    0.005383 -0.044642 -0.036385  0.021872  0.003935  0.015596  0.008142   
..        ...       ...       ...       ...       ...       ...       ...   
437  0.041708  0.050680  0.019662  0.059744 -0.005697 -0.002566 -0.028674   
438 -0.005515  0.050680 -0.015906 -0.067642  0.049341  0.079165 -0.028674   
439  0.041708  0.050680 -0.015906  0.017282 -0.037344 -0.013840 -0.024993   
440 -0.045472 -0.044642  0.039062  0.001215  0.016318  0.015283 -0.028674   
441 -0.045472 -0.044642 -0.073030 -0.081414  0.083740  0.027809  0.173816   

           s4        s5        s6  
0   -0.002592  0.019908 -0.017646  
1  

### 1.5) Print the target of the diabetes data (10 points)

In [7]:
print(diabetes_data.target)


[151.  75. 141. 206. 135.  97. 138.  63. 110. 310. 101.  69. 179. 185.
 118. 171. 166. 144.  97. 168.  68.  49.  68. 245. 184. 202. 137.  85.
 131. 283. 129.  59. 341.  87.  65. 102. 265. 276. 252.  90. 100.  55.
  61.  92. 259.  53. 190. 142.  75. 142. 155. 225.  59. 104. 182. 128.
  52.  37. 170. 170.  61. 144.  52. 128.  71. 163. 150.  97. 160. 178.
  48. 270. 202. 111.  85.  42. 170. 200. 252. 113. 143.  51.  52. 210.
  65. 141.  55. 134.  42. 111.  98. 164.  48.  96.  90. 162. 150. 279.
  92.  83. 128. 102. 302. 198.  95.  53. 134. 144. 232.  81. 104.  59.
 246. 297. 258. 229. 275. 281. 179. 200. 200. 173. 180.  84. 121. 161.
  99. 109. 115. 268. 274. 158. 107.  83. 103. 272.  85. 280. 336. 281.
 118. 317. 235.  60. 174. 259. 178. 128.  96. 126. 288.  88. 292.  71.
 197. 186.  25.  84.  96. 195.  53. 217. 172. 131. 214.  59.  70. 220.
 268. 152.  47.  74. 295. 101. 151. 127. 237. 225.  81. 151. 107.  64.
 138. 185. 265. 101. 137. 143. 141.  79. 292. 178.  91. 116.  86. 122.
  72. 

## 2) Split into Training and Testing data and print the training and testing data (50 points)

### 2.1) Split into Training and Testing data with training to testing ratio 80:20 (10 points)

In [16]:
from sklearn.model_selection import train_test_split

x_train,x_test,y_train,y_test = train_test_split(diabetes_data.data, diabetes_data.target, test_size=0.2)

### 2.2) print x_train (10 points)

In [17]:
print(x_train)

[[-0.06000263  0.05068012 -0.04716281 ... -0.03949338 -0.06291295
  -0.05492509]
 [ 0.00538306  0.05068012  0.03043966 ... -0.03949338  0.00864028
   0.01549073]
 [-0.00551455 -0.04464164  0.05630715 ...  0.03799897  0.05078151
   0.0569118 ]
 ...
 [ 0.03444337  0.05068012 -0.00943939 ...  0.07120998  0.0110081
  -0.02178823]
 [-0.00188202  0.05068012 -0.02452876 ... -0.00259226 -0.02139368
   0.03620126]
 [ 0.04534098 -0.04464164 -0.0191397  ... -0.03949338  0.01776348
  -0.01350402]]


### 2.3) Print y_train (10 points)

In [18]:
print (y_train)


[ 72. 202. 272.  73. 175.  63.  42.  70. 104. 237. 212.  67.  88.  48.
 259. 142. 232.  97. 275. 131. 258. 144. 142. 265. 168. 141.  91. 131.
 297.  71. 252.  97.  37. 128. 121. 182. 292. 198. 129.  53.  88.  59.
 179. 242.  88. 124.  64. 229. 206. 268. 264. 283. 148.  63.  72. 174.
  90. 237.  51. 171.  67. 220. 196.  96.  64. 214. 107. 235. 261. 177.
 292.  99. 170. 191. 153. 160. 281. 102. 242. 317.  44.  42. 191. 109.
  78. 154. 225. 321. 185. 137. 181.  71. 346. 118.  55. 196. 235.  97.
 190.  86. 243. 162. 131. 135. 198. 281. 147. 236.  60. 248. 273.  72.
 156. 202.  45.  80. 151. 155. 310. 192.  25.  85.  95. 116.  77. 233.
  39. 163. 109. 125.  66. 293.  88. 245.  83. 181. 241.  51.  85. 158.
  87.  90.  72. 139. 122.  61. 262. 279.  85.  89. 129. 144. 200.  94.
 308.  85.  49. 210. 200. 178. 268.  94.  51. 150. 303. 259. 170. 187.
 197.  72. 144. 195. 186.  83. 246.  68.  99. 145. 230. 185. 253. 107.
  39. 155. 219. 310. 178.  72. 172. 270.  69.  63. 217. 114.  71.  55.
 113. 

### 2.4) Print x_test (10 points)

In [21]:
print (x_test)


[[-9.63280163e-02 -4.46416365e-02 -7.62637389e-02 -4.35421882e-02
  -4.55994513e-02 -3.48207628e-02  8.14208361e-03 -3.94933829e-02
  -5.94726974e-02 -8.39198358e-02]
 [-5.27375548e-02 -4.46416365e-02  7.13965152e-02 -7.45280244e-02
  -1.53284884e-02 -1.31387743e-03  4.46044580e-03 -2.14118336e-02
  -4.68794828e-02  3.06440941e-03]
 [-7.81653240e-02 -4.46416365e-02 -1.69840749e-02 -1.25563519e-02
  -1.93006962e-04 -1.35266674e-02  7.07299263e-02 -3.94933829e-02
  -4.11803852e-02 -9.22040496e-02]
 [ 7.44012909e-02 -4.46416365e-02  3.15174685e-02  1.01058381e-01
   4.65893902e-02  3.68902349e-02  1.55053592e-02 -2.59226200e-03
   3.36568129e-02  4.44854786e-02]
 [-4.18399395e-02 -4.46416365e-02 -4.93184371e-02 -3.66564468e-02
  -7.07277125e-03 -2.26079728e-02  8.54564775e-02 -3.94933829e-02
  -6.64881482e-02  7.20651633e-03]
 [-1.88201653e-03 -4.46416365e-02 -3.85403164e-02  2.18723550e-02
  -1.08893283e-01 -1.15613066e-01  2.28686348e-02 -7.63945038e-02
  -4.68794828e-02  2.37749440e-02

### 2.5) Print y_test (10 points)

In [20]:
print (y_test)


[134.  55.  90. 296. 128.  40. 184. 281. 277. 197.  87. 202. 121. 128.
 123. 103. 116.  68. 127. 248. 113. 124. 170.  90. 152. 101. 168.  48.
 141. 140. 220.  53. 128.  61.  63. 242.  95.  77. 146. 178.  71. 101.
  74. 257. 140. 283. 259. 135. 276. 122. 215.  65.  84. 118. 150. 104.
  79. 201. 138.  92. 167. 182. 182. 143.  77. 272. 258. 168. 265. 139.
 101.  93.  59. 100. 118.  77. 206. 142. 190. 132. 275. 336. 185.  49.
 208.  48. 108. 200. 202.]
